.github/workflows/ci.yml

```diff
@@ -0,0 +1,161 @@
+        with:
+          path: |
+            ~/.cache/pip
+            ~/.cache/
+            ~/.ccache/
+          key: ${{ runner.os }}-py${{ matrix.python-version }}-${{ matrix.scikit-version }}
+
+      - name: Checkout repo
+        uses: actions/checkout@v2
+
+      - name: Set up ccache
+        run: |
+          sudo apt-get install -yq ccache
+          echo /usr/lib/ccache >> $GITHUB_PATH
+
+      - name: Install dependencies
+        run: |
+          pip install -U pip setuptools wheel twine
+          pip install numpy cython  # scikit-learn build deps if wheel missing
+          pip install scikit-learn==${{ matrix.scikit-version }}
+          pip install .[dev,plots]
+
+      - name: Python sdist
+        run: |
+          python setup.py sdist
+          twine check dist/*
+
+      - name: Test
+        if: env.TEST_WITH_COVERAGE != 'true'
+        run: |
+          python -c 'import skopt; skopt.show_versions()'
+          make test-code-parallel
+
+      - name: Test with Coverage
+        if: env.TEST_WITH_COVERAGE == 'true'
+        run: |
+          python -c 'import skopt; skopt.show_versions()'
+          make test-coverage
+          bash <(curl -s https://codecov.io/bash)
+
+      - name: Publish 📦 on PyPI
+        if: env.DO_PUBLISH == 'true'
+        uses: pypa/gh-action-pypi-publish@master
+        with:
+          password: ${{ secrets.PYPI_API_TOKEN }}
+
+  docs:
+    name: Docs
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        python-version: [ 3.9 ]
+    env:
+      MPLBACKEND: "agg"
+
+    steps:
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Set up caches
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-py${{ matrix.python-version }}
+
+      - name: Checkout repo
+        uses: actions/checkout@v2
+
+      - name: Install latex dependencies
+        run: |
+          sudo apt-get -yq update
+          sudo apt-get -yq --no-install-recommends install \
+              dvipng texlive-latex-base texlive-latex-extra \
+              texlive-latex-recommended texlive-fonts-recommended \
+              latexmk tex-gyre gsfonts ccache
+
+      - name: Install dependencies
+        run: |
+          pip install -U pip setuptools wheel
+          pip install -e .[dev,doc,plots]
+
+      - name: Test docs
+        run: |
+          python -c 'import skopt; skopt.show_versions()'
+          make test-doc
+          make test-sphinxext
+          make -C doc doctest
+          make -C doc linkcheck
+
+      - name: Build docs
+        run: |
+          .github/scripts/build_docs.sh
+
+      - name: Publish docs
+        if: env.DO_PUBLISH_DOCS == 'true'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_DOCS_TOKEN }}
+        run: |
+          .github/scripts/publish_docs.sh doc/_build/html/stable
```

Comment:
And this secret would need to be set to push generated docs to [the docs repo](https://github.com/scikit-optimize/scikit-optimize.github.io).